### PR TITLE
Add alias `.to_file()` for `.to()` method of structures and molecules

### PR DESCRIPTION
**Pull request by QuantumChemist**
## Summary

I have noticed that many beginners (including myself a while ago) find it very unintuitive to write e.g. structures to a file by using .to(), especially with the .from_file() method counterpart. That's why I added aliases `to_file` for structures and molecules.

PS: This is the second attempt of #3354 as something went wrong there...

### CODE REVIEW

File: pymatgen/core/structure.py [hunk 1]
```diff
@@ -2713,6 +2713,10 @@ def to(self, filename: str = "", fmt: str = "", **kwargs) -> str:
             writer.write_file(filename)
         return str(writer)
 
+    def to_file(self, filename: str = "", fmt: str = "", **kwargs) -> str:
+        """An alias for .to() for more user-friendliness."""
+        return self.to(filename, fmt, **kwargs)
```
Comment (janosh) [inline pymatgen/core/structure.py:2718]: I think we can move this to `SiteCollection` so only need a single alias and forward both `*args` and `**kwargs`?

**Comment by shyuep** [comment]
You don't actually need to write it this way. All that needs to be done is to add `to_file = to` and `write_file = to` in SiteCollection. That will create the aliases without so much code.

**Comment by QuantumChemist** [comment]
> You don't actually need to write it this way. All that needs to be done is to add `to_file = to` and `write_file = to` in SiteCollection. That will create the aliases without so much code.

I know, that's how I started, but when I tried to call the alias in pycharm, it wouldn't give me the input hints and at least for me they are really important. If you think that's not necessary, then I can change it to the simple alias.

**Review by janosh — CHANGES_REQUESTED** [review]
We definitely need a test for the new alias.

**Comment by QuantumChemist** [comment]
> We definitely need a test for the new alias.

Aye aye! 🫡

**Comment by QuantumChemist** [comment]
hi @janosh I hope the tests are ok like that?

**Review by janosh — APPROVED** [review]
Thanks @QuantumChemist! 👍

**Pull request merged**